server/tests/db/test_connection_pool.py

```
@@ -254,6 +254,183 @@ def test_putconn_failure_swallowed(self, fresh_pool):
+        ):
+            with pool.get_connection() as conn:
+                assert conn is connection
+
+        assert _RESET_SQL in _executed_sql(cursor)
+        factory.return_value.putconn.assert_called_once_with(connection)
+
+    # ------------------------------------------------------------------
+    # Direct RLS-table query with no context returns zero rows
+    # ------------------------------------------------------------------
+
+    def test_rls_protected_query_without_context_returns_no_rows(self, monkeypatch):
```

Ruff (0.15.12)

[warning] Unused method argument: `monkeypatch`

(ARG002)